### PR TITLE
kernel: add support for XMC XM25QH128C SPI-NOR

### PR DESCRIPTION
**Pull request by lyq1996**
Verified on jdcloud_re-cp-02.

Signed-off-by: lyq1996 <lyq123233@gmail.com>

XM25QH128C Datasheet: https://www.semiee.com/file/XMC/XMC-XM25QH128C.pdf

### CODE REVIEW

File: target/linux/generic/pending-5.10/485-mtd-spi-nor-add-xmc-xm25qh128c.patch [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+Signed-off-by: Kyle Liu <lyq123233@gmail.com>
+
+--- a/drivers/mtd/spi-nor/xmc.c
++++ b/drivers/mtd/spi-nor/xmc.c
+@@ -14,6 +14,8 @@ static const struct flash_info xmc_parts
+ 			    SECT_4K | SPI_NOR_DUAL_READ | SPI_NOR_QUAD_READ) },
+ 	{ "XM25QH128A", INFO(0x207018, 0, 64 * 1024, 256,
+ 			     SECT_4K | SPI_NOR_DUAL_READ | SPI_NOR_QUAD_READ) },
++	{ "XM25QH128C", INFO(0x204018, 0, 64 * 1024, 256,
++                             SECT_4K | SPI_NOR_DUAL_READ | SPI_NOR_QUAD_READ) },
```
Comment (ynezz) [inline target/linux/generic/pending-5.10/485-mtd-spi-nor-add-xmc-xm25qh128c.patch:14]: Some whitespace issue here?

**Comment by ynezz** [comment]
You've put it into `pending` directory, but I can't find it on upstream mailing lists. This needs to be sent upstream first.

File: target/linux/generic/pending-5.10/485-mtd-spi-nor-add-xmc-xm25qh128c.patch (hunk 1 above)
Comment (lyq1996) [reply target/linux/generic/pending-5.10/485-mtd-spi-nor-add-xmc-xm25qh128c.patch:14]: Yes, I will change it

**Comment by lyq1996** [comment]
> You've put it into `pending` directory, but I can't find it on upstream mailing lists. This needs to be sent upstream first.

Sorry ynzze, I'm newbee for submit patch, I'm current working on send this patch to Linux.

File: target/linux/generic/pending-5.10/485-mtd-spi-nor-add-xmc-xm25qh128c.patch (hunk 1 above)
Comment (ynezz) [reply target/linux/generic/pending-5.10/485-mtd-spi-nor-add-xmc-xm25qh128c.patch:14]: While at it, fix your `Signed-off-by: lyq1996 <lyq123233@gmail.com>`  as in https://openwrt.org/submitting-patches#sign_your_work

> using your real name (no pseudonyms or anonymous contributions.) "

**Comment by hauke** [comment]
@lyq1996 When you submitted the patch upstream please post a link to the patchwork or mailing list post here.

**Pull request closed without merging**